be/src/pipeline/pipeline_x/pipeline_x_task.cpp

```diff
@@ -326,6 +304,46 @@ Status PipelineXTask::execute(bool* eos) {
     return status;
 }
 
+bool PipelineXTask::should_revoke_memory(RuntimeState* state, int64_t revocable_mem_bytes) {
+    auto* query_ctx = state->get_query_ctx();
+    bool need_revoke = query_ctx->query_mem_tracker->need_revoke();
+    if (!need_revoke) {
+        return false;
+    }
+
+    const auto min_revocable_mem_bytes = state->min_revocable_mem();
+
+    auto tg = query_ctx->workload_group();
```

Comment:
If tg == nullptr then return false; to avoid core when there is no workload group bind to this query. And print log in this case.

Reply:
do not use tg. tg is old name "task group", use wg